fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""Utility class for plotting metrics with error ranges."""
+
+from logging import error
+from typing import Any, Dict, List, Union
+import pandas as pd
+
+from matplotlib.axes._axes import Axes
+
+from ._metric_frame import MetricFrame
+
+_MATPLOTLIB_IMPORT_ERROR_MESSAGE = "Please make sure to install matplotlib to use " \
+                                   "the plotting for MetricFrame."
+
+_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR = "Ambiguous Error Metric. Please only provide one of: error_bars and conf_intervals."
+
+_METRICS_NOT_LIST_OR_STR_ERROR = "Metric should be a string of a single metric or a list of the metrics in provided MetricFrame, but {0} was provided"
+_NEG_ERROR_OR_FLIPPED_BOUNDS_ERROR = "upper_error, lower_error, and symmetric_error must be positive and upper_bound must be greater than lower_bound"
+_INVALID_ERROR_MAPPING = "Invalid error mapping, each metric must have all the key-value pairs that make up an Error Format"
+
+def plot_metric_frame(plot_type: str, metric_frame: MetricFrame, 
+                    metrics: Union[List[str], str]=None, error_bars: Union[List[str], str]=None, conf_intervals: Union[List[str], str]=None,
+                    ax=None, show_plot=True, plot_error_labels: bool=True,
+                    text_precision_digits: int=4, text_fontsize: int=8, text_color: str="black", text_ha: str="center",
+                    title=None, capsize=10, colormap="Pastel1", TEMP_plot_with_df=True):
+    """Visualization for metrics with statistical error bounds.
```

Comment:
i suggest dropping any properties that can be adjusted on `ax` level; or add `**kwargs` and pass all kwargs to ax object:
https://github.com/scikit-learn/scikit-learn/blob/15a949460dbf19e5e196b8ef48f9712b72a3b3c3/sklearn/metrics/_plot/roc_curve.py#L80

Reply:
In the community call didn't we say to keep it "leaner" and not use **kwargs? I like the usage of **kwargs but I thought we said to only allow for users to edit things through changing `ax`